record/type.go

```diff
@@ -21,48 +20,40 @@ type TypeFields struct {
 	Labels      []string `json:"labels"`
 }
 
-// TypeClassification specifies a class for an Type
-type TypeClassification string
+// TypeName specifies a supported type name
+type TypeName string
 
-// list of valid classifications
 var (
-	TypeClassificationResource TypeClassification = "resource"
-	TypeClassificationDataset  TypeClassification = "dataset"
-	TypeClassificationSchema   TypeClassification = "schema"
-	TypeClassificationMetric   TypeClassification = "metric"
+	TypeNameTable     TypeName = "table"
+	TypeNameJob       TypeName = "job"
+	TypeNameDashboard TypeName = "dashboard"
+	TypeNameTopic     TypeName = "topic"
 )
 
-// AllTypeClassifications holds a list of valid classifications
-var AllTypeClassifications = []TypeClassification{
-	TypeClassificationResource,
-	TypeClassificationDataset,
-	TypeClassificationSchema,
-	TypeClassificationMetric,
+// String cast TypeName to string
+func (tn TypeName) String() string {
+	return string(tn)
 }
 
-// Type represents a named collection of records
-// Entities are supposed to represent resources, datasets and schema.
-// XXX(Aman): should Type names be case insensitive?
+// Type represents a typename wrapped in a JSON
 type Type struct {
-	Name           string             `json:"name"`
-	Classification TypeClassification `json:"classification"`
+	Name TypeName `json:"name"`
```

Comment:
Do we still need Type struct or we can operate with enum only?

Reply:
Unless we want to get rid of meta index, we need this type struct to add a new type in meta index.

Reply:
we can get rid of `_meta` index, we will not use it anymore.